### PR TITLE
snippet: document

### PR DESCRIPTION
**Pull request by kou**
It's not cover all features.

### CODE REVIEW

**Comment by kou** [comment]
This uses #1270 as the parent branch. So we should merge #1270 before this.

File: doc/source/reference/functions/snippet.rst [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+..   --query "fast performance"
+
+It returns only one snippet because ``"max_n_results": 1`` is specified.
+
+The max size of a text part is 200byte by default. The unit is bytes
+not characters. The size doesn't include inserted ``[`` and ``[``. You
+can change it by ``width`` option:
+
+.. groonga-command
+.. include:: ../../example/reference/functions/snippet/usage_max_n_results.log
```
Comment (HashidaTKS) [inline doc/source/reference/functions/snippet.rst:140]: Is this example file correct?
Also, it seems that the `width` option isn't used at this usage in the built html.

File: doc/source/reference/functions/snippet.rst [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+..                             })' \
+..   --match_columns content \
+..   --query "fast performance"
+
+You can detect snippet delimiter with regular expression by
+``delimiter_regexp`` option. You can use ``\.\s*`` to use only text in
+the target sentence. Note that you need to escape ``\`` in string:
+
+.. groonga-command
+.. include:: ../../example/reference/functions/snippet/usage_max_n_results.log
```
Comment (HashidaTKS) [inline doc/source/reference/functions/snippet.rst:156]: Is this example file correct?

File: doc/source/reference/functions/snippet.rst (hunk 1 above)
Comment (kou) [reply doc/source/reference/functions/snippet.rst:140]: Good catch!

File: doc/source/reference/functions/snippet.rst (hunk 2 above)
Comment (kou) [reply doc/source/reference/functions/snippet.rst:156]: Good catch!

**Pull request merged**